### PR TITLE
fix: Fix type definitions

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

**Comment by taion** [comment]
this is so our front end can successfully pass type checking again :p

File: .babelrc.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 module.exports = api => ({
   presets: [
     [
-      '@4c/4catalyzer',
+      '@4c',
       {
         target: 'web',
-        runtime: true,
+        targets: {},
```
Comment (taion) [inline .babelrc.js:7]: since this is in principle public

File: .eslintrc [hunk 2]
```diff
@@ -1,27 +1,11 @@
...
   "rules": {
     "prettier/prettier": "error"
-  },
-  "env": {
-    "browser": true
-  },
-  "overrides": [
-    {
-      "files": ["test/**"],
-      "plugins": ["jest"],
```
Comment (taion) [inline .eslintrc:13]: we don't have any tests here

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
   only:
     - master
+
+jobs:
+  include:
+    - stage: release
+      node_js: "10"
+      script:
+        - yarn run build
+        - yarn semantic-release
```
Comment (taion) [inline .travis.yml:21]: copied from @restart/context

File: package.json [hunk 4]
```diff
@@ -1,88 +1,85 @@
 {
   "name": "@4c/layout",
   "version": "0.2.1",
+  "files": [
+    "lib",
```
Comment (taion) [inline package.json:5]: there was so much junk on npm lol

File: src/css-modules/Block.js [hunk 5]
```diff
@@ -1,23 +1,27 @@
...
 function Block(
   {
-    className,
-    style,
-    inline,
-    alignSelf,
+    as: Component,
+    direction: _d,
+    pad: _p,
+    wrap: _w,
```
Comment (taion) [inline src/css-modules/Block.js:15]: we weren't previously stripping out the other flex props

File: src/css-modules/Flex.js [hunk 6]
```diff
@@ -1,26 +1,27 @@
...
   {
-    className,
-    style,
+    as: Component,
     direction,
     pad,
-    grow,
-    flex,
     wrap,
+    grow,
```
Comment (taion) [inline src/css-modules/Flex.js:16]: alphabetized to match the prop types

File: types/index.d.ts [hunk 7]
```diff
@@ -1,36 +1,31 @@
...
-export type ReplaceProps<Inner extends React.ReactType, P> = Omit<
+type ReplaceProps<Inner extends React.ElementType, P> = Omit<
   PropsOf<Inner>,
   P
 > &
   P;
 
-export interface BsPrefixProps<As extends React.ReactType> {
-  as?: As;
-  bsPrefix?: string;
```
Comment (taion) [inline types/index.d.ts:25]: @itajaja ಠ_ಠ

File: types/index.d.ts [hunk 8]
```diff
@@ -56,24 +51,45 @@ type Content =
...
+
 declare class Layout<
-  As extends React.ReactType = 'div'
-> extends BsPrefixComponent<As, LayoutProps> {}
+  As extends React.ElementType = 'div'
+> extends ComponentWithAs<As, LayoutProps> {
+  static Flex: typeof Flex;
+  static Block: typeof Block;
+
+  static Spacer: React.ComponentType;
```
Comment (taion) [inline types/index.d.ts:92]: we were using this

File: types/test.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
+// TypeScript Version: 3.2
+
+import * as React from 'react';
+import Layout from '@4c/layout';
+
+<Layout align="start" grow />;
+<Layout display="block" grow />;
+<Layout display="block" align="start" />; // $ExpectError
```
Comment (taion) [inline types/test.tsx:8]: this only errors against 3.2, which is why we specify 3.2 rather than 3.0 as elsewhere

File: types/index.d.ts (hunk 7 above)
Comment (itajaja) [reply types/index.d.ts:25]: lol 🤦‍♂️

File: types/index.d.ts [hunk 10]
```diff
@@ -56,24 +51,45 @@ type Content =
...
   inline?: boolean;
   flex?: boolean | number;
   /** default to 'stretch' */
-  align?: Align;
   alignSelf?: Align;
+}
+
+export interface FlexProps extends BlockProps {
+  /** default to 'row' */
+  direction?: 'row' | 'column';
```
Comment (itajaja) [inline types/index.d.ts:64]: I'd make it a type

File: types/index.d.ts [hunk 11]
```diff
@@ -56,24 +51,45 @@ type Content =
...
 
+declare class Flex<
+  As extends React.ElementType = 'div'
+> extends ComponentWithAs<As, FlexProps> {}
+
+declare class Block<
+  As extends React.ElementType = 'div'
+> extends ComponentWithAs<As, BlockProps> {}
+
+export type LayoutProps =
```
Comment (itajaja) [inline types/index.d.ts:82]: what is this wizardry? display is not even in the proptypes

**Review by itajaja — APPROVED** [review]
cool, so the parts that would break were basically the `static` things? Thanks for making this better

File: package.json (hunk 4 above)
Comment (itajaja) [reply package.json:5]: we should add it on the base cli

Comment (taion) [reply package.json:5]: i think we generally want to publish from `/lib` in many cases, though

File: types/index.d.ts (hunk 11 above)
Comment (taion) [reply types/index.d.ts:82]: yeah it is: https://github.com/4Catalyzer/layout/blob/483bc697065561048f3f092a908a4b5004fc5db6/src/css-modules/index.js#L9-L20

File: types/index.d.ts (hunk 10 above)
Comment (taion) [reply types/index.d.ts:64]: i was just matching https://github.com/4Catalyzer/layout/blob/master/src/PropTypes.js#L30

Comment (itajaja) [reply types/index.d.ts:64]: up to you

File: types/index.d.ts (hunk 11 above)
Comment (itajaja) [reply types/index.d.ts:82]: ah there it is, I was looking in the proptypes file

**Pull request merged**